### PR TITLE
Refactor the project

### PR DESCRIPTION
**Pull request by datalek**

### CODE REVIEW

File: src/domain/useCases/ConfirmConsentUseCase.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  (
+    interaction: Interaction,
+    rememberGrant: boolean
+  ): TE.TaskEither<DomainError, Grant> => {
+    if (ConsentResult.is(interaction.result)) {
+      return pipe(grantService.find(interaction.result.grantId), fromTEOtoTE);
+    } else if (LoginResult.is(interaction.result)) {
+      return TE.right({
+        expireAt: new Date(new Date().getTime() + 1000 * 60 * 60 * 24),
+        id: crypto.randomBytes(20).toString("hex") as GrantId,
```
[the remark this answers is not in the document]
Comment (datalek) [reply src/domain/useCases/ConfirmConsentUseCase.ts]: Done!

File: src/adapters/http/clients/index.ts [hunk 2]
```diff
@@ -11,12 +11,16 @@ import {
...
+  ServiceId,
+} from "../../../domain/clients/types";
+import { ClientService } from "../../../domain/clients/ClientService";
+import { Logger } from "../../../domain/logger";
 
 const makeAPIClientDetail = (input: Client): APIClientDetail => ({
-  application_type: input.applicationType,
+  application_type: "web",
   client_id: input.clientId,
   client_id_issued_at: input.issuedAt.toString(),
```
[the remark this answers is not in the document]
Comment (datalek) [reply src/adapters/http/clients/index.ts]: Done!

File: src/config.ts [hunk 3]
```diff
@@ -22,17 +22,9 @@ interface Info {
...
-interface Config {
-  readonly info: Info;
-  readonly IOBackend: IOBackend;
-  readonly server: ServerConfig;
-  readonly logger: logger.LogConfig;
-  readonly postgres: postgres.PostgresConfig;
-}
-
-type ConfEnv = NodeJS.ProcessEnv;
+type Envs = NodeJS.ProcessEnv;
```
[the remark this answers is not in the document]
Comment (datalek) [reply src/config.ts:23]: Done 👍

File: src/domain/utils.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { pipe } from "fp-ts/lib/function";
+import { Option } from "fp-ts/lib/Option";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
```
[the remark this answers is not in the document]
Comment (datalek) [reply src/domain/utils.ts]: Done 👍

File: prisma/schema.prisma [hunk 5]
```diff
@@ -1,109 +1,58 @@
...
-}
-
-model GrantToRemember {
-  accountId   String
-  clientId    String
-  grantId     String
-  grant       Grant @relation(fields: [grantId], references: [id])
-  @@unique([accountId, clientId])
+  id                String @id @default(auto()) @map("_id") @db.ObjectId
+  clientId          String @unique
```
Comment (francescopersico) [inline prisma/schema.prisma:12]: Why we are using this case type instead of "client_id"?

Comment (francescopersico) [inline prisma/schema.prisma:12]: Why we have both an id and a clientId?

**Pull request merged**